### PR TITLE
feat(cli): add a global `--context`

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml_cli/cloud.py [hunk 1]
```diff
@@ -71,18 +69,33 @@ def login(endpoint: str, api_token: str, context: str) -> None:  # type: ignore
...
+            )
+        )
 
     @cloud.command()
-    @click.argument("context", type=click.STRING)
-    def update_current_context(context: str) -> None:  # type: ignore (not accessed)
+    @click.argument("context_name", type=click.STRING)
+    def update_current_context(context_name: str) -> None:  # type: ignore (not accessed)
         """Update current context"""
-        ctx = CloudClientConfig.get_config().set_current_context(context)
```
Comment (aarnphm) [inline src/bentoml_cli/cloud.py:85]: To avoid colliding namespace of `context`

File: src/bentoml_cli/cloud.py [hunk 2]
```diff
@@ -60,7 +58,9 @@ def login(endpoint: str, api_token: str, context: str) -> None:  # type: ignore
             raise CLIException("current organization is not found")
 
         ctx = CloudClientContext(
-            name=context,
+            name=shared_options.context
```
Comment (sauyon) [inline src/bentoml_cli/cloud.py:61]: Er, is this right?

Comment (aarnphm) [reply src/bentoml_cli/cloud.py:61]: yes right, before the default value is `default_context_name`. now if the context is not passed, the default value is the same as b4

Comment (aarnphm) [inline src/bentoml_cli/cloud.py:61]:
```suggestion
            name=shared_options.cloud_context
```

**Review by sauyon — APPROVED** [review]
👍

**Pull request merged**